subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/InstantExecutionHost.kt

```diff
@@ -136,6 +136,8 @@ class InstantExecutionHost internal constructor(
             }
         }
 
+        override val gradle: GradleInternal = this@InstantExecutionHost.gradle
```

Comment:
:x: How about moving the `GradleInternal` requirement to the constructor and getting rid of this property and the `this@...` expression above?